src/app/modules/companies/store/companies.reducer.ts

```diff
@@ -0,0 +1,64 @@
+import { EditCompany } from './../../../models/edit-company.model';
+import { PartialDeep } from 'type-fest';
+import { createReducer, on } from '@ngrx/store';
+import * as CompaniesActions from './companies.actions';
+
+export const companiesFeatureKey = 'companies';
+
+export interface State {
+  selectedCompany: PartialDeep<EditCompany> | null,
+  editCompany: PartialDeep<EditCompany> | null,
+}
+
+export const initialState: State = {
+  selectedCompany: null,
+  editCompany: null,
+};
+
+export const reducer = createReducer(
+  initialState,
+
+  on(CompaniesActions.getCompany, (state) : State => state),
+  on(CompaniesActions.getCompanySuccess, (state, action): State  =>  {
+    const organization = action.data.organization;
+    return {
+      ...state,
+      selectedCompany: organization === null
+        ? null
+        : {
+          id: organization.id,
+          name: organization.name as string,
+        },
```

Comment:
Similar code is repeated here 3 times. May be move it to (helper) function/class method?